nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/GetKinesisStream.java

```diff
@@ -0,0 +1,474 @@
+                context.yield();
+            }
+        }
+    }
+
+    @OnUnscheduled
+    public void stopConsuming(final ProcessContext context) {
+        final AtomicBoolean success = new AtomicBoolean(true);
+        final Stream<Map.Entry<Worker, Future<?>>> workerStream;
+        if (workers.size() > 1) {
+            workerStream = workers.entrySet().parallelStream();
+        } else {
+            workerStream = workers.entrySet().stream();
+        }
+        workerStream.forEach(entry -> {
+            if (!shutdownWorker(entry.getKey(), entry.getValue())) {
+                success.set(false);
+            }
+        });
+
+        executorService.shutdown();
+        workers = null;
+
+        if (!success.get()) {
+            getLogger().warn("One of more problems while shutting down Kinesis Workers, see logs for details");
+            context.yield();
+        }
+    }
+
+    private Worker prepareWorker(final ProcessContext context, final ProcessSession session) {
+        final String appName = getApplicationName(context);
+        final String streamName = getStreamName(context);
+        final String workerId = generateWorkerId();
+        final long checkpointIntervalMillis = getCheckpointIntervalMillis(context);
+
+        final KinesisRecordProcessorFactory factory = new KinesisRecordProcessorFactory(session, getLogger(),
+                getClient().getEndpointPrefix(), checkpointIntervalMillis, retryWaitMillis, numRetries, dateTimeFormatter);
+
+        final KinesisClientLibConfiguration kinesisClientLibConfiguration = prepareKinesisClientLibConfiguration(
+                context, appName, streamName, workerId);
+
+        final Worker.Builder workerBuilder = prepareWorkerBuilder(kinesisClientLibConfiguration, factory, context);
+
+        getLogger().info("Kinesis Worker prepared for application {} to process stream {} as worker ID {}...",
+                appName, streamName, workerId);
+
+        return workerBuilder.build();
+    }
+
+    KinesisClientLibConfiguration prepareKinesisClientLibConfiguration(final ProcessContext context, final String appName,
+                                                                       final String streamName, final String workerId) {
+        final AWSCredentialsProvider credentialsProvider = new AWSStaticCredentialsProvider(awsCredentials);
+
+        @SuppressWarnings({"deprecated", "java:S1874"})
+        // use most of the defaults in the constructor chain rather than the mammoth constructor here
+        final KinesisClientLibConfiguration kinesisClientLibConfiguration =
+                new KinesisClientLibConfiguration(appName, streamName, credentialsProvider, workerId)
+                        .withCommonClientConfig(getClient().getClientConfiguration());
+
+        kinesisClientLibConfiguration.withRegionName(getRegion().getName())
+                .withMaxInitializationAttempts(numRetries);
```

Comment:
Would be better to allow use of Dynamic Properties to set this and other Kinesis Client Library Configuration properties. There are a lot of KCL config settings that could be tweaked and allowing a user to set them (without increasing the processor's already rather large list of static properties) would seem sensible.

I'll look to push a change for this in the near future.

Reply:
Second commit includes the update to allow dynamic property use for configuring the KCL